swim/test_utils.go

```diff
@@ -146,6 +146,23 @@ func bootstrapNodes(t *testing.T, testNodes ...*testNode) {
 		})
 		require.NoError(t, err, "node must bootstrap successfully")
 	}
+
+	if waitForConvergence {
```

Comment:
Nice one! Can we do this using public interfaces?

I realise that no disseminator changes should mean that the ring has converged, but that is an internal implementation detail.

I think we should be looping over the nodes in the ring to ensure they all have the same checksum, as this matches our definition of "converged".

You mentioned yesterday that it would be ideal to call a `ProtocolPeriod` during some tests. Note that can easily be triggered by calling the `/admin/gossip/tick` endpoint.

Finally, I think it would be good to have a timeout here, i.e. if something goes wrong and convergence doesn't occur within a healthy timeframe for the test (e.g. 500ms), then we fail to avoid hanging indefinitely.